wallet-new/server/Main.hs

```diff
@@ -118,19 +118,21 @@ startEdgeNode WalletStartupOptions{..} = do
 -- the reason why we don't generate a yaml file is because for swagger-ui is actually
 -- much better to start with the JSON input, as the tool is capable of generating
 -- better-looking YAMLs.
-generateSwaggerDocumentation :: IO ()
+generateSwaggerDocumentation :: (HasCompileInfo, HasUpdateConfiguration) => IO ()
 generateSwaggerDocumentation = do
     BL8.writeFile "wallet-new/spec/swagger.json" (encodePretty Swagger.api)
     putText "Swagger API written on disk."
 
 -- | The main entrypoint for the Wallet.
 main :: IO ()
 main = withCompileInfo $(retrieveCompileTimeInfo) $ do
-  cfg <- getWalletNodeOptions
-  putText "Wallet is starting..."
-  generateSwaggerDocumentation
-  let loggingParams = CLI.loggingParams loggerName (wsoNodeArgs cfg)
-  loggerBracket loggingParams . runProduction $ do
-    CLI.printFlags
-    logInfo "[Attention] Software is built with the wallet backend"
-    startEdgeNode cfg
+    cfg@WalletStartupOptions{..} <- getWalletNodeOptions
+    putText "Wallet is starting..."
+    let loggingParams = CLI.loggingParams loggerName wsoNodeArgs
+        conf = CLI.configurationOptions $ CLI.commonArgs wsoNodeArgs
+    loggerBracket loggingParams . runProduction $ do
+        withConfigurations conf $ do
+            liftIO generateSwaggerDocumentation
```

Comment:
As a consequence we have to wait 2 secs longer until wallet starts for swagger doc to be written.

Alternatively I can do something like  : 
```haskell
loggerBracket loggingParams $ withConfigurationOptions conf generateSwaggerDocumentation
loggerBracket loggingParams $ runProduction $ ...
```
cons of the above is that we will parse configuration two times. 

thoughts ?

Reply:
@akegalj What about generating the Swagger docs only if `RunMode` is `DebugMode` so that we avoid compromising performances? Then, once we have [this PR](https://github.com/input-output-hk/cardano-sl/pull/2068) fully merged, I think we can finally move away from generating the Swagger docs in `Main`. Would that work as an interim measure? 😉

Reply:
god point. Will fix